libs/community/langchain_community/chains/azure_content_safety_chain.py

```diff
@@ -0,0 +1,163 @@
+"""Pass input through an azure content safety resource."""
+
+from typing import Any, Dict, List, Optional
+
+from langchain.chains.base import Chain
+from langchain_core.callbacks import (
+    CallbackManagerForChainRun,
+)
+from langchain_core.exceptions import LangChainException
+from langchain_core.utils import get_from_dict_or_env
+from pydantic import model_validator
+
+
+class AzureHarmfulContentError(LangChainException):
+    """Exception for handling harmful content detected
+    in input for a model or chain according to Azure's
+    content safety policy."""
+
+    def __init__(
+        self,
+        input: str,
+    ):
+        """Constructor
+
+        Args:
+            input (str): The input given by the user to the model.
+        """
+        self.input = input
+        self.message = "The input has breached Azure's Content Safety Policy"
+        super().__init__(self.message)
+
+
+class AzureAIContentSafetyChain(Chain):
+    """
+    A wrapper for the Azure AI Content Safety API in a Runnable form.
+    Allows for harmful content detection and filtering before input is
+    provided to a model.
+
+    **Note**:
+    This Service will filter input that shows any sign of harmful content,
+    this is non-configurable.
+
+    Attributes:
+        error (bool): Whether to raise an error if harmful content is detected.
+        content_safety_key (Optional[str]): API key for Azure Content Safety.
+        content_safety_endpoint (Optional[str]): Endpoint URL for Azure Content Safety.
+
+    Setup:
+        1. Follow the instructions here to deploy Azure AI Content Safety:
+            https://learn.microsoft.com/azure/ai-services/content-safety/overview
+
+        2. Install ``langchain`` ``langchain_community`` and set the following
+        environment variables:
+
+        .. code-block:: bash
+
+            pip install -U langchain langchain-community
+
+            export AZURE_CONTENT_SAFETY_KEY="your-api-key"
+            export AZURE_CONTENT_SAFETY_ENDPOINT="https://your-endpoint.azure.com/"
+
+
+    Example Usage:
+        .. code-block:: python
+
+            from langchain_community.chains import AzureAIContentSafetyChain
+            from langchain_openai import AzureChatOpenAI
+
+            moderate = AzureAIContentSafetyChain()
+            prompt = ChatPromptTemplate.from_messages([("system",
+                    "repeat after me: {input}")])
+            model = AzureChatOpenAI()
+
+            moderated_chain = moderate | prompt | model
+
+            moderated_chain.invoke({"input": "Hey, How are you?"})
+    """
+
+    client: Any = None  #: :meta private:
+    error: bool = False
+    """Whether or not to error if bad content was found."""
```

Comment:
I would default to True. If a user was going through the trouble of adding this to their chain, it feels to me it's because they want to react to harmful content, i.e. do something if an exception is raised